Source/Dafny/Rewriter.cs

```diff
@@ -118,6 +119,36 @@ public abstract class IRewriter {
       Contract.Requires(program != null);
     }
   }
+  // intern project 
+
+  class tobiRewriter : IRewriter {
+    internal override void PostResolve(Program program) {
+      base.PostResolve(program);
+      foreach (var moduleDefinition in program.Modules()) {
+        foreach (var topLevelDecl in moduleDefinition.TopLevelDecls.OfType<TopLevelDeclWithMembers>()) {
+          foreach (var function in topLevelDecl.Members.OfType<Function>()) {
+            if (function.Body is NestedMatchExpr matchExpr) {
+              var idPatterns = matchExpr.Cases.Select(matchCase => matchCase.Pat).OfType<IdPattern>();
+              foreach (var idPattern in idPatterns) {
+                if (char.IsLower(idPattern.Id[0]) == false && idPattern.Arguments == null) {
+                  var lineNum = idPattern.Tok.line;
+                  string message = "Variable name '" + idPattern + "' on line " + lineNum +
```

Comment:
You can make this easier to read by using an interpolated string.

Also, I wouldn't put the location in the message since it's already passed to the `.Warning(` call which'll use it for some form of visualisation.